tests/integration/test_log_rotation.py

```diff
@@ -0,0 +1,164 @@
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_log_rotation(ops_test: OpsTest, mysql_router_charm_series: str) -> None:
+    """Test the log rotation of mysqlrouter logs."""
+    # Build and deploy applications
+    mysqlrouter_charm = await ops_test.build_charm(".")
+
+    # deploy mysqlrouter with num_units=None since it's a subordinate charm
+    # and will be installed with the related consumer application
+    applications = await asyncio.gather(
+        ops_test.model.deploy(
+            "mysql",
+            channel="8.0/stable",
+            application_name=MYSQL_APP_NAME,
+            config={"profile": "testing"},
+            num_units=1,
+        ),
+        ops_test.model.deploy(
+            mysqlrouter_charm,
+            application_name=MYSQL_ROUTER_APP_NAME,
+            num_units=None,
+        ),
+        ops_test.model.deploy(
+            APPLICATION_APP_NAME,
+            application_name=APPLICATION_APP_NAME,
+            num_units=1,
+            # MySQL Router is subordinate—it will use the series of the principal charm
+            series=mysql_router_charm_series,
+            channel="latest/stable",
+        ),
+    )
+
+    mysql_app, mysql_router_app, application_app = (
+        applications[0],
+        applications[1],
+        applications[2],
+    )
+    unit = application_app.units[0]
+
+    await ops_test.model.relate(
+        f"{MYSQL_ROUTER_APP_NAME}:backend-database", f"{MYSQL_APP_NAME}:database"
+    )
+
+    # the mysqlrouter application will be in unknown state since it is a subordinate charm
+    async with ops_test.fast_forward():
+        await asyncio.gather(
+            ops_test.model.block_until(lambda: mysql_app.status == "active", timeout=SLOW_TIMEOUT),
+            ops_test.model.block_until(
+                lambda: application_app.status == "waiting", timeout=SLOW_TIMEOUT
+            ),
+        )
+
+        await ops_test.model.relate(
+            f"{MYSQL_ROUTER_APP_NAME}:database", f"{APPLICATION_APP_NAME}:database"
+        )
+
+        await asyncio.gather(
+            ops_test.model.block_until(lambda: mysql_app.status == "active", timeout=SLOW_TIMEOUT),
+            ops_test.model.block_until(
+                lambda: mysql_router_app.status == "active", timeout=SLOW_TIMEOUT
+            ),
+            ops_test.model.block_until(
+                lambda: application_app.status == "active", timeout=SLOW_TIMEOUT
+            ),
+        )
+
+    logger.info("Removing the cron file")
+    await delete_file_or_directory_in_unit(
+        ops_test, unit.name, "/etc/cron.d/flush_mysqlrouter_logs"
+    )
+
+    logger.info("Stopping any running logrotate jobs")
+    await stop_running_flush_mysqlrouter_cronjobs(ops_test, unit.name)
+
+    logger.info("Removing existing archive directory")
+    await delete_file_or_directory_in_unit(
+        ops_test,
+        unit.name,
+        f"{CHARMED_MYSQL_COMMON_DIRECTORY}/var/log/mysqlrouter/archive_mysqlrouter/",
+    )
```

Comment:
we are unable to test the cron job because we could not robustly ensure that cron would not run while we run `logrotate`. so the test is disabling the cron part and just testing that the `logrotate` command works correctly

we could revisit this test to improve it to also test cron